satellytes/satellytes.com
# feat: add cookie consent

This PR adds a cookie banner to satellytes.com. With this banner, the user can accept cookies for YouTube Embeds and [Leadinfo](https://www.leadinfo.com/en/). 

## Why
This is necessary because leadinfo uses cookies to track better which companies visit the website (currently it runs without cookies). The youtube embeds in blog posts need to be accepted by the user because the user has to accept the useage of youtube cookies and the youtube ToS. 

## How
To easily manage cookies and to help making it GDPR conform [CookieConsent](https://github.com/orestbida/cookieconsent) is used. CookieConsent also provides an easy way to enable scripts based on cookie settings which allows for easy activation of the leadinfo script when accepted.
For the youtube embeds the CookieConsent plugin [IFrameManager](https://github.com/orestbida/iframemanager/tree/main) is used for a simple GDPR compliant solution.

To implement the cookie banner code from the [CookieConsent docs](https://cookieconsent.orestbida.com/essential/introduction.html) was used. The docs also contain usage examples of how to set up IFrameManager which was used in combination with the [README of IFrameManger](https://github.com/orestbida/iframemanager/tree/main#) to implement the youtube embeds.

The legal texts were copied from Ipeks PR (#825). 

### Changes:
- add `vanilla-cookieconsent` module (cookie and script management)
- add `@orestbida/iframemanager` module (youtube embeds)
- add cookie consent popup
- manage youtube emmbed consent into cookie consent popup 
- add leadinfo consent to cookie consent 

## How to test
1. Go on any page of the [website](https://deploy-preview-830--satellytes.netlify.app/) and see if the cookie banner appears 
2. Decline all cookies to test if the no cookies are added (except the cookie consent cookie that stores that all cookies were declined, which requires no consent because it is an essential cookie) 
3. Check if YouTube embeds are disabled by looking at a blog post with youtube embeds 
4. Check if enabling youtube for one embed works ("Load content")
5. Check if enabling youtube for all embeds works ("Load content and don't ask again")
6. Reset cookie selection by deleting the cookie consent cookie
7. Reload the page and select "Accept all" 
8. Check if all youtube embeds work
9. Check if leadinfo has placed cookies (cookie may not be placed in every session) 
10. Reset cookie selection again
11. Test enabling only one of the cookie categories 
12. Test if it worked correctly by checking as described above 

## Screenshots
<details> 
<summary>Screenshots of the Cookie Banner</summary>

![image](https://github.com/user-attachments/assets/60e880a3-072f-4073-a0e2-cf1cd2bb094b)
![image](https://github.com/user-attachments/assets/789e4c55-c483-4ba0-8c49-3a0d6e3e79c4)

</details> 

<details> 
<summary>Screenshot of the IFrameManger consent overlay</summary>

![image](https://github.com/user-attachments/assets/86138b06-bc39-47b7-a565-7cdfa1273e20)

</details> 

<details> 
<summary>Screenshot of Cookies when all cookies are declined</summary>

![image](https://github.com/user-attachments/assets/f16f0cc5-a3ac-4836-9c47-6a95af0b1e5d)
</details> 


<details> 
<summary>Screenshots of Cookies when only YoutTube embeds are accepted</summary>

![image](https://github.com/user-attachments/assets/39b5d7b1-3073-4043-ae1d-d1011f64a584)
![image](https://github.com/user-attachments/assets/46ece9e5-c6e8-424f-81ca-b18478000838)

</details> 

<details> 
<summary>Screenshots of Cookies when only Leadinfo embeds are accepted</summary>

![image](https://github.com/user-attachments/assets/ae4badc5-42c8-415a-9cce-d214da5e2990)

</details> 

## Note:
After merging this setting has to be disbled in the [Leadinfo Portal](https://portal.leadinfo.com/settings/tracker/LI-66D184CC97CFD)
![image](https://github.com/user-attachments/assets/08a0b6e1-1f8a-46a4-a8c6-1da4d2918537)
